app/models/event.rb

```diff
@@ -85,6 +85,23 @@ def calculate_round_robin
     save
   end
 
+  def calculate_double_round_robin
+    pairings_per_day = round_robin_pairings(teams.to_a)
+    pairings_per_day += round_robin_pairings(teams.to_a)
+    pairings_per_day.each_with_index do |day, gameday|
+      day.each do |pairing|
+        # Creating a match for every pairing if one of the teams is nil (which happens if there is an odd number of teams)
+        # the other team will have to wait for this day
+        if gameday < (teams.to_a.size / 2)
+          matches << Match.new(team_home: pairing[0], team_away: pairing[1], gameday: gameday + 1) unless pairing[0].nil? or pairing[1].nil?
```

Comment:
the nil? check should be included in the `round_robins_pairings` method. So that it only return valid pairings. That should reduce code duplication